src/api/video.ts

```diff
@@ -0,0 +1,28 @@
+import Bugsnag from "@bugsnag/js";
+import firebase from "firebase/app";
+
+export interface GetVideoTokenProps {
+  userId: string;
+  roomName: string;
+}
+
+export const getVideoToken = async ({
+  userId,
+  roomName,
+}: GetVideoTokenProps): Promise<void | firebase.functions.HttpsCallableResult> => {
+  return firebase
+    .functions()
+    .httpsCallable("video-getToken")({
+      identity: userId,
+      room: roomName,
+    })
+    .catch((err) => {
+      Bugsnag.notify(err, (event) => {
+        event.addMetadata("context", {
+          location: "api/video::getVideoToken",
+          userId,
+          roomName,
+        });
+      });
+    });
```

Comment:
I don't think we should rethrow them. Since we don't currently have a standardized way to handle it, 

I would assume that we would do something like `onError` along with calling the `show UI-friendly message`.

Reply:
Done in https://github.com/sparkletown/sparkle/pull/1299/commits/cba5c35d7853b6001ca1b1bcb342e9415fa0ddec

Reply:
> I don't think we should rethrow them. Since we don't currently have a standardized way to handle it

If we continue to hide it away like this, we never will. Bugsnag informs us of errors, it does nothing to communicate errors to users. When things randomly break/don't work for seemingly no reason (because we didn't communicate that something went wrong), that is terrible UX, and leads to all sorts of issues/complaints.

It's pretty trivial to use the async helpers from react-use that I mentioned in another comment on this PR for neatly wrapping Promises, and handling there responses, including error state. Then when the error state is set, we just display that in the UI. We don't need a big fancy overengineered solution to things when the most basic of things will handle it for us.

---

> I would assume that we would do something like onError along with calling the show UI-friendly message.

That's why I said to rethrow the errors within the `Promise`, so that the downstream consuming code can use `.catch` and handle it. There's no point re-inventing the wheel with legacy patterns (`onError` callback) when `Promise` provides everything we need in a better/more usable way. The node/JS ecosystem moved from callback based API's to using Promise-based API's for a reason.

Reply:
> Done in cba5c35

As described above, there is no point adding callback functions like this to a `Promise` based API. That's the whole point of `Promise`.

Reply:
Same problem in https://github.com/sparkletown/sparkle/pull/1299#discussion_r626278802